core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceMethod.java

```diff
@@ -499,6 +499,17 @@ private static String httpHeaderValue(Parameter entry, HttpRequest req) {
         return entryDefaultValue(entry);
     }
 
+    @SuppressWarnings("unchecked")
```

Comment:
@trustin, @minwoox, @hyangtack, @anuraaga Thanks you very much for many comments :) BTW, I'm curious about that if the user wants to use case-sensitive enum parameter, then should we need to keep the map for the enumeration type? (Map<String, Enum<?>>) I considered about adding the map inside the Parameter class, but how about this implementation just checking when the convert to string? (maybe this is more simpler?)

Reply:
Hmm.. I also thought the time complexity would be O(n), the performance degradation is so high in the case of a lot of the enum element Let me change to the map as @trustin suggested.

Reply:
@AmosDoan Yes, please. We may want to introduce some conversion option or value converters to support case-sensitive enums, but for now, let's keep it case-insensitive.